### PR TITLE
Adding the build publish steps.

### PR DESCRIPTION
**Pull request by VinayakGadad**
## Purpose
- https://sift.atlassian.net/browse/CI-13163

## Summary
- The job installs dependencies, builds, and pushes a gem to RubyGems if the version does not exist already.
- It extracts the gem version from lib/sift/version.rb using awk.
- The job also runs Rake tasks, such as listing tasks, building, and installing the gem.

## Testing
- There is no release at the moment, Once this is ready, we can uncomment.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -81,8 +81,41 @@ jobs:
...
+            sudo gem install bundler
+            sudo bundle install
+      - run:
+          name: Build and push gem
+          command: |
+            version=$(awk -F'"' '/ VERSION = /  {print $2}' < lib/sift/version.rb)
+            if [[ -z $(gem list -r -e sift -v $version --no-verbose) ]]; then
+                echo "Gem version does not exists on RubyGems."
+                gem build sift.gemspec
+                gem push sift-$version.gem
```
Comment (viaskal-sift) [inline .circleci/config.yml:102]: seems, this will be done on every commit, right?

File: .circleci/config.yml [hunk 2]
```diff
@@ -81,8 +81,44 @@ jobs:
...
+            sudo bundle exec rake build
+            sudo bundle exec rake install
+#             sudo bundle exec rake release - Uncomment this line when release is available.
+
 workflows:
   ruby-test:
     jobs:
       - build:
           context: *context
+      - build_publish:
```
Comment (viaskal-sift) [inline .circleci/config.yml:119]: but this one will publish on every commit on master. We would like to have this process under control

Comment (VinayakGadad) [reply .circleci/config.yml]: I changed this to use the github action workflow to do the releases.

**Review by andycai-sift — APPROVED** [review]
LGTM

**Pull request merged**